Kernel/Devices/SerialDevice.h

```diff
@@ -106,25 +108,49 @@ class SerialDevice final : public CharacterDevice {
         DataReady = 0x01 << 0
     };
 
+    // ^File
+    virtual bool can_write(const FileDescription&, size_t) const override;
+
     // ^Device
     virtual mode_t required_mode() const override { return 0620; }
     virtual String device_name() const override;
 
+    // ^TTY
+    virtual String const& tty_name() const override { return m_tty_name; }
+
+    static Optional<speed_t> termios_baud_from_serial(Baud);
+    static Optional<Baud> serial_baud_from_termios(speed_t);
+
 private:
-    friend class PCISerialDevice;
+    // ^TTY
+    virtual ssize_t on_tty_write(const UserOrKernelBuffer&, ssize_t) override;
+    virtual void echo(u8) override;
+    virtual int change_baud(speed_t in_baud, speed_t out_baud) override;
+    virtual int change_parity(TTY::ParityMode) override;
+    virtual int change_stop_bits(TTY::StopBits) override;
+    virtual int change_character_size(TTY::CharacterSize) override;
+    virtual int change_receiver_enabled(bool) override;
+    virtual int change_ignore_status_lines(bool) override;
+    virtual void discard_pending_input() override;
+    virtual void discard_pending_output() override;
+    virtual void wait_until_pending_output_completes() override;
 
     // ^CharacterDevice
     virtual const char* class_name() const override { return "SerialDevice"; }
 
+    // ^IRQHandler
+    virtual void handle_irq(const RegisterState&) override;
+
     void initialize();
```

Comment:
this method doesnt exist anymore

Reply:
What do you mean?

Reply:
you removed the initialize method...?

Reply:
Oh. I thought that the `handle_irq` was the issue because of the git highlight. Weird that it compiles though.

Reply:
theres nothing illegal about declaring unimplemented methods as long as you dont use them, it just looks weird :)